hphp/runtime/vm/jit/region-tracelet.cpp

```diff
@@ -404,6 +405,20 @@ void recordDependencies(Env& env) {
     if (type <= TCls) return;
     auto& whichMap = hint ? hintMap : guardMap;
     auto inret = whichMap.insert(std::make_pair(loc, type));
+    // Unconstrained pseudo-main guards will be relaxed to Gen by the guard
+    // relaxation pass. Since we don't allow loading TGen locals
+    // in pseudo-main, save the predicted type here.
+    if (guard->marker().func()->isPseudoMain()) {
+      auto const pred = RegionDesc::TypedLocation {loc, type};
```

Comment:
Also, the opt builds are failing because we have `-Werror=unused-but-set-variable` enabled internally, and, because `FTRACE` is a `do { } while (0)` in opt builds, `pred` ends up never being referenced.

The easiest solution is probably to just construct the `TypedLocation` inline as you pass it to `show` down below.